test/extension/test.sh

```diff
@@ -651,17 +670,163 @@ EOF
     sql cdb_testmember_1 "SELECT cartodb.CDB_SetUp_User_PG_FDW_Table('another_user_defined_test', 'test_fdw', 'foo');"
     sql postgres "SELECT cartodb._CDB_Drop_User_PG_FDW_Server('another_user_defined_test', /* force = */ true)"
 
+    tear_down_fdw_target
+}
 
-    # Teardown
-    DATABASE=fdw_target sql postgres 'REVOKE USAGE ON SCHEMA test_fdw FROM fdw_user;'
-    DATABASE=fdw_target sql postgres 'REVOKE SELECT ON test_fdw.foo FROM fdw_user;'
-    DATABASE=fdw_target sql postgres 'REVOKE SELECT ON test_fdw.foo2 FROM fdw_user;'
-    DATABASE=fdw_target sql postgres 'REVOKE SELECT ON cdb_tablemetadata_text FROM fdw_user;'
-    DATABASE=fdw_target sql postgres 'DROP ROLE fdw_user;'
 
-    sql postgres "select pg_terminate_backend(pid) from pg_stat_activity where datname='fdw_target';"
-    DATABASE=fdw_target tear_down_database
+function test_federated_tables() {
+    setup_fdw_target
+
+    # Federated server configuration for tests
+    read -d '' federated_server_config <<- EOF
+{
+    "server": {
+        "dbname": "fdw_target",
+        "host": "localhost",
+        "port": ${PGPORT:-5432}
+    },
+    "credentials": {
+        "username": "fdw_user",
+        "password": "foobarino"
+    }
 }
+EOF
+
+    # Unit-test __ft_credentials_to_user_mapping
```

Comment:
`__ft_credentials_to_user_mapping` needs more tests to exemplify what's the expected input and output